azurerm/internal/services/eventhub/eventhub_data_source.go

```diff
@@ -0,0 +1,80 @@
+package eventhub
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func dataSourceEventHub() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceEventHubRead,
+
+		Timeouts: &schema.ResourceTimeout{
+			Read: schema.DefaultTimeout(5 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"namespace_name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"resource_group_name": azure.SchemaResourceGroupNameForDataSource(),
+
+			"partition_count": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+
+			"partition_ids": {
+				Type:     schema.TypeSet,
```

Comment:
since this is a Data Source, this'll need to be a List rather than a Set (since otherwise these values won't be accessible) - as such can we make this:

```suggestion
				Type:     schema.TypeList,
```